### DISCUSSION TITLE
input.rtmp not reading from rtmp remote source but opening rtmp service

### DISCUSSION THREAD

**Discussion by kessenza** [savonet/liquidsoap#2200] [User Support]
call input.rtmp is expected to work getting remote source from rtmp external source (like input.ffmpeg and input.http do) but instead it opens a rtmp service locally (like input.harbor does)

For instance if I have an ngnix with rtmp module running locally if I launch
input.rtmp("rtmp://my-local-ngnix/endpoint") it conflicts with existing open port managed by ngnix

If I stop ngnix the above script runs smoothly

I am running Liquidsoap 2.0.2 on Ubuntu 20 installed via opam

**Comment by toots** [comment]
I have just added a parameter to allow to set `input.rtmp` as listener or client, it will be coming out with the `2.0.3` release.

Meanwhile, `input.rtmp` is a simple wrapper around `input.ffmpeg` so you can just drop this in your script:
```ruby
def input.rtmp(~max_buffer=5., ~listen=true, url)
  input.ffmpeg(max_buffer=max_buffer, log_overfull=false, format="live_flv",
               int_args=[("listen", listen ? 1 : 0)], url)
end
```
